src/main/scala/ai/privado/languageEngine/go/tagger/collection/CollectionTagger.scala

```diff
@@ -0,0 +1,173 @@
+package ai.privado.languageEngine.go.tagger.collection
+
+import ai.privado.cache.RuleCache
+import ai.privado.model.{CatLevelOne, Constants, InternalTag, Language, NodeType, RuleInfo}
+import ai.privado.tagger.PrivadoParallelCpgPass
+import io.shiftleft.codepropertygraph.generated.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes.Method
+import io.shiftleft.semanticcpg.language.*
+import ai.privado.utility.Utilities.*
+
+import scala.collection.immutable.HashMap
+import scala.collection.mutable
+import scala.util.{Failure, Success, Try}
+
+class CollectionTagger(cpg: Cpg, ruleCache: RuleCache) extends PrivadoParallelCpgPass[RuleInfo](cpg) {
+  private val methodUrlMap            = mutable.HashMap[Long, String]()
+  private val classUrlMap             = mutable.HashMap[Long, String]()
+  private val ROUTES_FILE_PATTERN     = ".*(routes|routers).go"
+  private val COLLECTION_FUNC_PATTERN = "(?i).*(Get|Post|Put|Patch|Delete).*"
+  private val COLLECTION_REST_PATTERN = "(?i)(Get|Post|Put|Patch|Delete)"
+
+  override def generateParts(): Array[RuleInfo] =
+    ruleCache.getRule.sources.toArray
+
+  override def runOnPart(builder: DiffGraphBuilder, sourceRuleInfo: RuleInfo): Unit = {
+    tagFuncCallCollection(builder, sourceRuleInfo)
+    tagRestCallCollection(builder, sourceRuleInfo)
+  }
+
+  private def tagFuncCallCollection(builder: DiffGraphBuilder, sourceRuleInfo: RuleInfo): Unit = {
+    val collectionCallMethod = cpg.call
+      .name("Methods")
+      .where(_.astChildren.isLiteral.code(COLLECTION_FUNC_PATTERN))
+      .astChildren
+      .isCall
+      .name("HandleFunc")
+      .l
+
+    // smaple route -> myrouter.HandleFunc("/user/", getUser).Method("GET")
+    val collectionMethodsCache = collectionCallMethod
+      .map { m =>
+        if (m.argument.size >= 2) {
+          val targetCollectionUrl =
+            m.argument(1).code
+
+          if (targetCollectionUrl.nonEmpty) {
+            val methodName             = m.argument(2).code
+            val targetCollectionMethod = cpg.method.name(methodName).l
```

Comment:
can you check if matching method on its `methodFullName` is possible instead of `name`?

Reply:
As discussed, `methodFullName` won't help, as we don't have complete method full name of method.